src/types/client-manager.ts

```diff
@@ -64,14 +98,63 @@ export class ClientManager {
     return this._kmd
   }
 
+  private _getNetworkPromise: Promise<SuggestedParamsWithMinFee> | undefined
+  /**
+   * Get details about the current network.
+   * @example Getting genesis ID
+   * ```typescript
+   * const network = await networkClient.network()
+   * const genesisId = network.genesisId
+   * ```
+   * @returns The current network details
+   */
+  public async network(): Promise<NetworkDetails> {
```

Comment:
This plus the following deprecates a series of legacy methods and has better performance since it caches the network params